lib/kauth/save.go

```diff
@@ -0,0 +1,45 @@
+package kauth
+
+import (
+	"fmt"
+	"github.com/enfabrica/enkit/lib/cache"
+	"github.com/enfabrica/enkit/lib/kcerts"
+	"github.com/enfabrica/enkit/lib/logger"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	"time"
+)
+
+var NoSSHCredentialsErr = errors.New("no credentials passed in")
+// SaveCredentials saves the passed in credentials to the current ssh-agent. If the credentials are empty, i.e.
+// the EnkitCredentials only contain EnkitCredentials.Token, it will return NoSSHCredentialsErr.
+func SaveCredentials(credentials *EnkitCredentials, store cache.Store, l logger.Logger) error{
+	if len(credentials.CaHosts) == 0 || credentials.SSHCertificate == nil || len(credentials.PrivateKey) == 0 {
+		return NoSSHCredentialsErr
```

Comment:
I would personally just return nil here. Not really a failure, we saved the nil credentials by not writing them anywhere. Makes the calling code simpler.

Also, no user of this library needs to know about this error. The only user ignores it. Add it if we actually ever end up needing it?

Reply:
Removed the error and updated for the latest pr